### ISSUE TITLE
ReactiveProperty without a default value

### ISSUE THREAD

**Issue by Tinche** [neuecc/UniRx#207]
Hello,

as far as I understand, there are two ways of initializing a ReactiveProperty:

* with a default value
* with a source observable

When given a default value, the property will propagate this value immediately to its subscribers. When given a source observable, the property will not propagate anything until the source observable produces something.

I find myself bridging my reactive properties with non-UniRx event systems, and I'd really like a way to create a reactive property without a source observable and without a default value, which will not propagate any values until something is first set to it.

I could probably define a helper subject, then initialize my reactive property with this subject and then set the value into the subject when it comes but it feels like a lot of ceremony for something that should be easy to do.

Think it'd be useful?

**Comment by FodderMK** [comment]
One way we get around that is to use ReplaySubject.  If you create a new ReplaySubject with a buffer size of 1 then it don't emit anything until the first OnNext call, and every OnNext call after that will send the newest value.

**Comment by Tinche** [comment]
Thanks, that's clever and we might use that. Would still prefer to see this in ReactiveProperty :)

**Comment by paveldk** [comment]
Agree. There should be option to disable emit if no default value was passed

**Comment by Ryo-Auredi** [comment]
See if you can get away with `SkipLatestValueOnSubscribe()`.

**Comment by nievesj** [comment]
I ended up commenting line 193 on ReactiveProperty.cs. There should be an option to disable that OnNext on subscribe, in fact it should be disabled by default. I can see situations where the current value is needed, but most of the time all I want is the next one.

`public IDisposable Subscribe(IObserver<T> observer)
        {
            if (isDisposed)
            {
                observer.OnCompleted();
                return Disposable.Empty;
            }

            //JMN: This is a non desired behavior. When subscribing to a ReactiveProperty it should not emit the current value
            // raise latest value on subscribe
            // observer.OnNext(value);

            // subscribe node, node as subscription.
            var next = new ObserverNode<T>(this, observer);
            if (root == null)
            {
                root = last = next;
            }
            else
            {
                last.Next = next;
                next.Previous = last;
                last = next;
            }
            return next;
        }`
